src/App.tsx

```diff
@@ -15,47 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT,
+  ALPHABET,
+  LENGHT,
+}
+
+interface Reorder {
+  sortField: SortType;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(goods: string[], { sortField, isReversed }: Reorder) {
+  const preparedGoods = [...goods];
+
+  if (sortField !== SortType.DEFAULT) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.LENGHT:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+  const goods = getPreparedGoods(goodsFromServer, { sortField, isReversed });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          onClick={() => setSortField(SortType.ALPHABET)}
+          className={cn('button', 'is-info',
+            { 'is-light': sortField !== SortType.ALPHABET })}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          onClick={() => setSortField(SortType.LENGHT)}
+          className={cn('button', 'is-info',
+            { 'is-light': sortField !== SortType.LENGHT })}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          onClick={() => setIsReversed(!isReversed)}
+          className={cn(
+            'button is-warning',
+            { 'is-light': !isReversed },
+          )}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(isReversed || sortField !== SortType.DEFAULT) && (
+          <button
+            onClick={() => {
+              setSortField(SortType.DEFAULT);
+              setIsReversed(false);
+            }}
```

Comment:
```suggestion
            onClick={reset}
```
Consider moving this logic from jsx